Added link to plot_adaboost_multiclass.py example

virchan opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs

Towards #26927 
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->


#### What does this implement/fix? Explain your changes.

Includes a link to the <code>plot_adaboost_multiclass.py</code> example


#### Any other comments?

NA


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

virchan committed:
Added link for plot_adaboost_multiclass example

marenwestermann commented:
Hi @virchan! You added a link to the example in the example itself. The links need to be placed in the doc strings of the functions/ estimators and if applicable the user guide. For an example see PR #26877.

virchan committed:
Moved the example link from the example itself back to the doc string
in _weight_boosting.py.

virchan committed:
Merge branch 'main' of https://github.com/virchan/scikit-learn

virchan commented:
Hello @marenwestermann,

Thank you for the reference, it's really helpful. I've moved the example link from the example itself to the doc string in <code>sklearn\ensemble\_weight_boosting.py</code>.

virchan committed:
Merge branch 'scikit-learn:main' into main

virchan committed:
Reworded the example reference of AdaBoost in the
`ensemble/_weight_boosting.py` file, moving it below the `Examples`
section for improved organization.

Included an AdaBoost example reference within the DecisionTree class in
the `tree/-class.py` file.

virchan committed:
Merge branch 'main' of https://github.com/virchan/scikit-learn

virchan commented on sklearn/ensemble/_weight_boosting.py:
```diff
@@ -351,6 +351,9 @@ class AdaBoostClassifier(
 
     Read more in the :ref:`User Guide <adaboost>`.
 
+    For more example of usage, see
```
Thank you for writing back!

The following files are modified:

**`ensemble\_weight_boosting.py`**: Reworked the example reference and moved it below the `Example` section.
**`tree\_classes.py`**: Added the example reference of using AdaBoost with decision trees for the `DecisionTree` class.

Please let me know if you need anything else.

virchan committed:
Merge branch 'scikit-learn:main' into main

marenwestermann reviewed (commented):
Apologies for the belated reply. I added a small comment (see below). Additionally it would be good to add a link to the example [to this section of the user guide](https://scikit-learn.org/stable/modules/tree.html#classification) which you can find in the code [here](https://github.com/scikit-learn/scikit-learn/blob/main/doc/modules/tree.rst?plain=1#L92). Just add the link to this example to the examples that are already listed there.

virchan committed:
Merge branch 'scikit-learn:main' into main

virchan committed:
- Added the Multi-class AdaBoosted Decision Trees example to the Decision Trees user guide.
- Modified the doc-string wording in the `AdaBoostClassifier` class referencing to the aforementioned example.

virchan committed:
Merge branch 'main' of https://github.com/virchan/scikit-learn

virchan committed:
Merge branch 'main' of https://github.com/virchan/scikit-learn

virchan commented:
Thank you for the reply. I have updated the following files:

- `sklearn\ensemble\_weight_boosting.py`: Corrected the doc-string referencing to the Adaboost example.
- `doc\modules\tree.rst`: Added the Adaboost example to the Decision Trees user guide.

Please let me know if I missed anything.

marenwestermann commented:
Thank you for the update @virchan. Can you fix the linting issue? When you click on the details under "ci/circleci: lint" in the code checks you'll see the error messages.

virchan committed:
Merge branch 'main' into main

virchan committed:
Reformatted doc-strings to meet the ruff requirement

virchan committed:
Merge branch 'main' of https://github.com/virchan/scikit-learn

virchan committed:
Empty commit for test re-run.

virchan commented:
Thank you for the follow-up!

The linting issue is fixed, all checks have passed.

adrinjalali commented on sklearn/ensemble/_weight_boosting.py:
```diff
@@ -478,6 +478,10 @@ class AdaBoostClassifier(
     >>> clf.score(X, y)
     0.96...
+
+    For a detailed example of using AdaBoost to fit a sequence of DecisionTrees
+    as weaklearners, please refer to
+    :ref:`sphx_glr_auto_examples_ensemble_plot_adaboost_multiclass.py`.
```
I think this is the only place where the mention makes sense. I would then remove the other two mentions in this PR.

adrinjalali reviewed (commented):
Otherwise LGTM.

virchan committed:
Merge branch 'scikit-learn:main' into main

virchan committed:
Merge branch 'scikit-learn:main' into main

virchan committed:
Removed example links from two files

virchan committed:
Merge branch 'main' of https://github.com/virchan/scikit-learn

virchan committed:
Empty commit for checks re-run

virchan committed:
Removed an empty line for checks re-run.

virchan commented:
Hello,

Thank you for the feedback! The example references are removed from the `doc\modules\tree.rst` and the `sklearn/tree/_classes.py` files.

Please let me know if there is anything else.
